### PR TITLE
Move message parsing out of the read loop

### PR DESCRIPTION
**Pull request by bdraco**
To further reduce the pressure on the esp, we want to read and drain the buffer as fast as possible, and since we have to wait to read while the data is processing we want the read loop to read as much as possible
first to keep the latency down and reduce the memory pressure on the esp.

### CODE REVIEW

File: aioesphomeapi/connection.py [hunk 1]
```diff
@@ -511,46 +511,47 @@ async def _report_fatal_error(self, err: Exception) -> None:
...
-                msg = await self._to_process.get()
+                pkt = await self._to_process.get()
             except RuntimeError:
                 break
 
+            msg_type = pkt.type
+            raw_msg = pkt.data
+            if msg_type not in MESSAGE_TYPE_TO_PROTO:
+                _LOGGER.debug("%s: Skipping message type %s", self.log_name, msg_type)
+                return
```
Comment (bdraco) [inline aioesphomeapi/connection.py:529]:
```suggestion
                continue
```

**Comment by bdraco** [comment]
I'll move the readexactly change to a new PR.

Went from 25000 to 9000 per minute so nice improvement

**Comment by bdraco** [comment]
the long uuid conversion have a double lower call

Screenshot 2022-11-30 at 8 57 53 AM

**Comment by bdraco** [comment]
Last push is much better
Screenshot 2022-11-30 at 9 06 17 AM

**Comment by bdraco** [comment]
Will move the message parsing optimizations to a new PR as well when I'm done going through all this

**Review by jesserockz — APPROVED** [review]

**Pull request merged**